Code refactoring (#3)
CodethinkLabs/elasticsearch-test-data: README.md

[I 241114 10:42:31 opensearch_test:47] ***Start Data Generate Test***
[I 241114 10:42:32 requests:83] Trying to create index http://localhost:9200/test_type
[I 241114 10:42:33 requests:89] Looks like the index exists already
[I 241114 10:42:33 upload_data:83] Generating 20000 docs, upload batch size is 10000
[I 241114 10:42:55 requests:109] Upload: OK - upload took:  4551ms, total docs uploaded:   10000
[I 241114 10:43:18 requests:109] Upload: OK - upload took:  4598ms, total docs uploaded:   20000
[I 241114 10:43:18 upload_data:123] Done - total docs uploaded: 20000, took 45 seconds
[I 241114 10:43:18 opensearch_test:52] ***Start Query All Test***
[I 241114 10:43:41 requests:226] Total hits: 20000, Total pages: 2
[I 241114 10:43:44 requests:231] Retrieved page 1 of 2
...
[I 241114 10:43:48 requests:231] Retrieved page 2 of 2
[I 241114 10:50:09 requests:213] Scroll context cleared successfully
[I 241114 10:50:09 requests:235] Total Querying time taken: 13116.00ms
[I 241114 10:50:10 opensearch_test:67] ***Start Delete Index***
[I 241114 10:50:13 requests:61] Deleting index 'test_type' done b'{"acknowledged":true}'

```
 

> Without any command line options, it will generate and upload 100000 documents

Reviewer: I would scale this back to 1000, and use a batch size of 1000 by default

}
```
to an Elasticsearch cluster at `http://localhost:9200` to an index called

> `test_type`.

Reviewer: why has this changed? the default index had a sensible name of `test_data`

## Not bad but what can I configure?

`python search-test.py --help` also gives you the full set of command line options,  
here are more description about the most important ones:

- `action`: [generate_data, query_all, custom_query, delete_index, all] choose one
  - generate_data: upload the data generated through `format` to the OpenSearch database.
  - query_all: request all values of the specified index within the range using `start_time` and `finish_time`.

>   - custom_query: You can specify the values for the body used in the request through a JSON file. this option require `json_path`. For more [read docs](https://opensearch.org/docs/latest/api-reference/search/)

Reviewer: ```suggestion
  - custom_query: You can specify the values for the body used in the request through a JSON file. this option require `json_path`. For more information read the api documentation - [ElasticSearch](add url here) [OpenSearch](https://opensearch.org/docs/latest/api-reference/search/)
```

  - delete_index: All data at the specified index will be deleted. (Please use with caution.)

>   - all: I will conduct whole process test.(generate_data -> query_all -> delete_index)

Reviewer: ```suggestion
  - all: (default) Conduct whole process test.(generate_data -> query_all -> delete_index)
```

> - The values that need to be set according to the server's security settings are as follows:
>   - `validate_cert` : SSL validate_cert for requests. Use false for self-signed certificates
>   - `username` : the username when basic auth is required
>   - `password` : the password when basic auth is required
>   - `client_cert` : (optional)filepath of CA certificates in PEM format
>   - `client_key` : (optional)filepath of for client SSL key

Reviewer: ```suggestion
- For authentication to the server, the following options are available:
  - `validate_cert` : SSL validate_cert for requests. Use false for self-signed certificates
  - Certificate based auth:
    - `client_cert` : filepath of CA certificates in PEM format
    - `client_key` : filepath of for client SSL key
  - Username based auth:
    - `username` : the username when basic auth is required
    - `password` : the password when basic auth is required
```

## Not bad but what can I configure?

> Recommended method for Config is create `server.conf` file and input the values needed.

Reviewer: I would re-order this back to the original layout:

* basic help description
* most common options
* list of all options
* config file setup

This would read better for new users who just want this to work

  given list of `-` seperated words, the words are optional defaulting to
  `text1` `text2` and `text3`, min and max are optional, defaulting to `1`
  and `1`
- `arr:[array_length_expression]:[single_element_format]` an array of entries
  with format specified by `single_element_format`. `array_length_expression`
  can be either a single number, or pair of numbers separated by `-` (i.e. 3-7),
  defining range of lengths from with random length will be picked for each array
  (Example `int_array:arr:1-5:int:1:250`)
- `log_version` a random version `str` looks like v1.1.1
- `sha` generate random sha(len 40)
- `file_name` Generate fake file `file_name:.py`  
- `uuid` Generate fake uuid
- `systemd` Generate fake systemd name

> ## How to setup config file

Reviewer: Move this section above the document format docs and have at a level 3 header
